.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,31 @@
+name: shellcheck
+on:
+  pull_request:  
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'     
+        required: true
+        default: 'warning'
+
+jobs:
+  shellcheck:
+    runs-on: gardenlinux-amd64
+    steps:
+    - uses: actions/checkout@v2
+    - run: |
+        echo "Install shellcheck"
+        echo "deb http://deb.debian.org/debian testing main contrib non-free" > /etc/apt/sources.list \
```

Comment:
on the github hosted runners you can then probably just do update & install shellcheck from default sources

Reply:
installs spellcheck 0.7.0

https://manpages.ubuntu.com/manpages/focal/man1/shellcheck.1.html